### PR TITLE
feat(validate-icu): add a cli to validates icu locales files

### PR DESCRIPTION
**Pull request by philibea**
Add a cli tools to helps us validate locales files translate by our docs team.

### CODE REVIEW

File: packages/validate-icu-locales/locales/en.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export default {
```
Comment (chambo-e) [inline packages/validate-icu-locales/locales/en.ts:1]: test purpose ?

File: rollup.config.mjs [hunk 2]
```diff
@@ -33,14 +33,15 @@ const getConfig = (pkg, isBrowser = false) => {
     },
```
Comment (chambo-e) [inline rollup.config.mjs:33]: Can be reverted

File: packages/validate-icu-locales/locales/en.ts (hunk 1 above)
Comment (philibea) [reply packages/validate-icu-locales/locales/en.ts]: Yepa

**Review by baptisteArno — APPROVED** [review]

File: packages/validate-icu-locales/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    }
+
+    if (extension === 'ts' || extension === 'js') {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+      const d: { default: Record<string, string> } = importFromStringSync(
+        fs.readFileSync(file).toString(),
+        { transformOptions: { loader: 'ts' } },
+      )
+
+      if (typeof d.default === 'object') {
```
Comment (QuiiBz) [inline packages/validate-icu-locales/src/index.ts:59]: I don't think the implicit cast and eslint-disable are necessary here.

```suggestion
      const mod = importFromStringSync(
        fs.readFileSync(file).toString(),
        { transformOptions: { loader: 'ts' } },
      )

      if ('default' in mod && typeof mod.default === 'object') {
```

File: packages/validate-icu-locales/src/__tests__/locales/en.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+export default {
+  //  error on this one missing bracket
+
+  'units.minutes.label':
+    '{count, plural, =0 {Minute} =1 {Minute} other {Minutes',
+  'units.hours.label': '{count, plural, =0 {Hour} =1 {Hour} other {Hours}}',
+  'units.days.label': '{count, plural, =0 {Day} =1 {Day} other {Days}}',
+  'units.months.label': '{count, plural, =0 {Month} =1 {Month} other {Months}}',
+}
```
Comment (QuiiBz) [inline packages/validate-icu-locales/src/__tests__/locales/en.ts:9]: We should add another test locale in TS with `as const` at the end.

**Review by matthprost — APPROVED** [review]

File: packages/validate-icu-locales/src/index.ts [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        value,
+        key,
+        filePath,
+      })
+    }
+  })
+}
+
+const readFiles = (files: string[]) =>
+  files.forEach(file => {
```
Comment (chambo-e) [inline packages/validate-icu-locales/src/index.ts:38]: async

File: packages/validate-icu-locales/src/index.ts [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+
+const { error, table } = console
+
+type Err = {
+  err: unknown
+  value: string
+  key: string
+  filePath: string
+}
+const errors: Err[] = []
```
Comment (chambo-e) [inline packages/validate-icu-locales/src/index.ts:18]: why use a global ?

**Review by chambo-e — CHANGES_REQUESTED** [review]

File: packages/validate-icu-locales/README.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+```bash
+$ pnpm add -D @scaleway/validate-icu-locales
+```
+## Usage
+
+We can parse JSON, TS and JS if theses files use default export.
+
+
+```
+node @scaleway/validate-icu-locales "../**/en.json"
```
Comment (chambo-e) [inline packages/validate-icu-locales/README.md:15]: why node?

File: packages/validate-icu-locales/src/index.ts [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+
+if (errors.length > 1) {
+  try {
+    throw new Error('There is somes ICU error')
+  } catch (err) {
+    error({ errors })
+    info(err)
+    process.exit(1)
+  }
+}
```
Comment (chambo-e) [inline packages/validate-icu-locales/src/index.ts:86]: The throw is kinda useless :)

File: packages/validate-icu-locales/README.md (hunk 7 above)
Comment (philibea) [reply packages/validate-icu-locales/README.md]: add rollup plugin to preserve shebang :)

File: packages/validate-icu-locales/src/index.ts (hunk 8 above)
Comment (philibea) [reply packages/validate-icu-locales/src/index.ts]: Yes 🙌  removed

File: rollup.config.mjs [hunk 9]
```diff
@@ -54,9 +53,14 @@ const getConfig = (pkg, isBrowser = false) => {
...
         browser: isBrowser,
-        extensions: [ '.mjs', '.js', '.json', '.ts', '.tsx' ],
+        extensions: ['.mjs', '.js', '.json', '.ts', '.tsx'],
         preferBuiltins: true,
       }),
+      preserveShebangs(),
+      // shebang({
+      //   include: ['**/index.js'],
+      //   shebang: '#!/usr/bin/env node'
+      // }),
```
Comment (chambo-e) [inline rollup.config.mjs:63]:
```suggestion
```

File: packages/validate-icu-locales/src/index.ts [hunk 10]
```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env node
+
+import { parse } from '@formatjs/icu-messageformat-parser'
+import { info } from 'console'
```
Comment (chambo-e) [inline packages/validate-icu-locales/src/index.ts:4]: Why use package and global ?

File: packages/validate-icu-locales/src/index.ts [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+      isICUError = true
+      error({
+        err,
+        value,
+        key,
+        filePath,
+      })
+    }
+  })
+}
```
Comment (chambo-e) [inline packages/validate-icu-locales/src/index.ts:31]: I don't understand why you are trying to complicate things 😄 

Why not return the error and combine them in the readFiles ? Then if the readFiles return an array it means there is error

File: packages/validate-icu-locales/src/index.ts [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+const files = await globby(pattern)
+
+if (files.length > 0) {
+  table(files)
+  await readFiles(files)
+}
+
+if (files.length === 0) {
+  info('There is no files matching this pattern', pattern)
+}
```
Comment (chambo-e) [inline packages/validate-icu-locales/src/index.ts:82]:
```suggestion
const files = await globby(pattern)

if (files.length === 0) {
  error('There is no files matching this pattern', pattern)
  process.exit(1)
}

table(files)
await readFiles(files)

```

File: rollup.config.mjs [hunk 13]
```diff
@@ -74,7 +74,7 @@ export default async () => {
   const doesAlsoTargetBrowser = 'browser' in pkg
 
   return [
-    getConfig(pkg),
+    getConfig(pkg, false),
```
Comment (chambo-e) [inline rollup.config.mjs:77]: Maybe remove the default then ?

File: packages/validate-icu-locales/package.json [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+  "name": "@scaleway/validate-icu-locales",
+  "version": "1.0.0",
+  "description": "A small cli to handle icu errors on locales files",
+  "keywords": [
+    "icu",
+    "i18n",
+    "cli"
+  ],
+  "type": "module",
+  "bin": "dist/index.js",
```
Comment (chambo-e) [inline packages/validate-icu-locales/package.json:11]: Maybe 

```suggestion
  "bin": {
    "validate-icu": "dist/index.js"
   },
```

A tad weird to have scoped packages as cli

Comment (philibea) [reply packages/validate-icu-locales/package.json:11]: agree

File: packages/validate-icu-locales/README.md [hunk 15]
```diff
@@ -0,0 +1,49 @@
...
+```bash
+$ pnpm add -D @scaleway/validate-icu-locales
+```
+## Usage
+
+We can parse JSON, TS and JS if theses files use default export.
+
+
+```
+@scaleway/validate-icu-locales "../**/en.json"
```
Comment (chambo-e) [inline packages/validate-icu-locales/README.md:18]:
```suggestion
validate-icu "../**/en.json"
```

File: packages/validate-icu-locales/src/index.ts [hunk 16]
```diff
@@ -0,0 +1,117 @@
...
+
+table(files)
+
+const errors = await readFiles(files)
+
+if (errors.length > 0) {
+  process.exit(1)
+}
+
+process.exit(0)
```
Comment (chambo-e) [inline packages/validate-icu-locales/src/index.ts:117]: No need, let node handle process termination :)

**Review by baptisteArno — APPROVED** [review]

File: packages/validate-icu-locales/README.md [hunk 17]
```diff
@@ -0,0 +1,49 @@
...
+
+```
+validate-icu "../**/en.json"
+```
+
+If there is an error on a local, the CLI will throw an Error and print all errors.
+
+
+## Error
+````
```
Comment (QuiiBz) [inline packages/validate-icu-locales/README.md:25]: Extra `

**Review by chambo-e — APPROVED** [review]

**Review by QuiiBz — APPROVED** [review]

**Pull request merged**